src/voteHelper/VoteHelper.ts

```diff
@@ -0,0 +1,131 @@
+import { ApiPromise } from '@polkadot/api';
+import { SubstrateConnection } from '../connection/SubstrateConnection';
+import { NetworkConfig } from '../config/conf';
+import { sendTransaction } from '../utils';
+import { Referendum } from './referendum';
+import { AbstainVote, StandardVote, TrackVotes, Vote } from './votes';
+
+export class VoteHelper {
+    private api: ApiPromise;
+
+    constructor(connection: SubstrateConnection) {
+        this.api = connection.getApi();
+    }
+
+    async checkVotes(network: NetworkConfig, sender): Promise<void> {
+        const ongoingReferena = await this.getOngoingReferenda();
+        const potentialVotes = await this.getPotentialVotes(network, ongoingReferena);
+
+        for (const validator of network.validators) {
+            const newVotes = await this.decideHowToVote(validator, potentialVotes);
+
+            for (const vote of newVotes) {
+                await this.voteForReferenda(vote, sender, validator, network.voteAmount);
+            }
+        }
+    }
+
+    async getOngoingReferenda(): Promise<Referendum[]> {
+        const referendums = await this.api.query.referenda.referendumInfoFor.entries();
+        const castedReferendums = referendums.map(([key, value]) => new Referendum(key, value));
+        return castedReferendums.filter(referendum => referendum.data.ongoing);
+    }
+
+    async getAccountVotes(accountAddress: string): Promise<Vote[]> {
+        const uniqueVotes = new Set<number>();
+        const referendaTracks = this.api.consts.referenda.tracks;
+        const trackArray = (referendaTracks as any).map(([key, value]) => new TrackVotes(key, value));
+        const accountVotes: Vote[] = [];
+
+        for (const track of trackArray) {
+            const votesInTrack = await this.api.query.convictionVoting.votingFor(accountAddress, track.trackNumber);
+            (votesInTrack as any).asCasting.votes.map((voteData) => {
+                const vote = Vote.createVote(voteData);
+                if (!uniqueVotes.has(vote.referendaNumber)) {
+                    uniqueVotes.add(vote.referendaNumber);
+                    accountVotes.push(vote);
+                }
```

Comment:
What are the resons for checking votes against uninquness by referenda? As far as I know it is not possible for an account to have two votes for the same referenda

Reply:
in that case I resolve priority of voting, in that loop I went through votersAccounts and save their votes, uniqueness need to save the first vote I found

Reply:
Yeah, but the thing you are talking about happens in `getPotentialVotes `.  Here, in `getAccountVotes` you fetch votes for a single account and we can be sure no duplicates will be there for a given referenda

Reply:
I've got you, thanks - fixed